File: 哈希表/442. 数组中重复的数据/数组中重复的数据.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;
// 时间复杂度O(n)，空间复杂度O(n)

class Solution
{
public:
    vector<int> findDuplicates(vector<int> &nums)
    {
        vector<int> arr;
        unordered_set<int> repetitivenumber;
        for (auto data : nums)
        {
            if (repetitivenumber.count(data))
            {
                arr.push_back(data);
            }
            repetitivenumber.insert(data);
        }
        return arr;
    }
};

File: 哈希表/442. 数组中重复的数据/数组中重复的数据.go
func findDuplicates(nums []int) []int {
	// 首先定义一个哈希表
	repetitivenumber := make(map[int]bool)
	var arr []int
	for _, value := range nums {
		_, ok := repetitivenumber[value]
		if ok {
			arr = append(arr, value)
		} else {
			repetitivenumber[value] = true
		}
	}
	return arr
}